### PR TITLE
MINOR: Bump trunk to 3.10.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by gharris1727**
Until KIP-853 is feature complete in a 3.x release, we are not yet ready breaking changes and a 4.0.0 version.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by jlprat — APPROVED** [review]
LGTM

**Review by chia7712 — APPROVED** [review]

File: tests/kafkatest/__init__.py [hunk 1]
```diff
@@ -22,4 +22,4 @@
 # Instead, in development branches, the version should have a suffix of the form ".devN"
 #
 # For example, when Kafka is at version 1.0.0-SNAPSHOT, this should be something like "1.0.0.dev0"
-__version__ = '4.0.0.dev0'
+__version__ = '3.9.0.dev0'
```
Comment (junrao) [inline tests/kafkatest/__init__.py]: Hmm, should this be 3.10?

**Review by junrao — COMMENTED** [review]
@gharris1727 : Thanks for the PR. Just a minor comment.

File: tests/kafkatest/__init__.py (hunk 1 above)
Comment (jlprat) [reply tests/kafkatest/__init__.py]: This always confuses me. I never know which version should go in there.

**Comment by chia7712** [comment]
According to the discussion (https://lists.apache.org/thread/tbpdy31jzybsz1chb2p8bdzv60znj329), 3.9 is the last 3 release. Maybe we should close this PR to avoid confusion

**Pull request closed without merging**